include/nuttx/can.h

```diff
@@ -63,10 +83,58 @@
  * frames.
  */
 
-#define CANFD_BRS 0x01 /* bit rate switch (second bitrate for payload data) */
-#define CANFD_ESI 0x02 /* error state indicator of the transmitting node */
+#define CANFD_BRS 0x01 /* Bit rate switch (second bitrate for payload data) */
+#define CANFD_ESI 0x02 /* Error state indicator of the transmitting node */
+#define CANFD_FDF 0x04 /* Mark CAN FD for dual use of struct canfd_frame */
+
+#define CAN_INV_FILTER 0x20000000u /* To be set in can_filter.can_id */
+
+/* PF_CAN protocols */
+
+#define CAN_RAW      1           /* RAW sockets */
+#define CAN_BCM      2           /* Broadcast Manager */
+#define CAN_TP16     3           /* VAG Transport Protocol v1.6 */
+#define CAN_TP20     4           /* VAG Transport Protocol v2.0 */
+#define CAN_MCNET    5           /* Bosch MCNet */
+#define CAN_ISOTP    6           /* ISO 15765-2 Transport Protocol */
+#define CAN_J1939    7           /* SAE J1939 */
+#define CAN_NPROTO   8
+
+#define SOL_CAN_BASE 100
+#define SOL_CAN_RAW  (SOL_CAN_BASE + CAN_RAW)
+
+/* CAN_RAW socket options */
+
+#define CAN_RAW_FILTER         (__SO_PROTOCOL + 0)
```

Comment:
`__SO_PROTOCOL` is defined in `sys/socket.h`, so we need to add `#include <sys/socket.h>`.

Reply:
Done！